### DISCUSSION TITLE
Does RegisterAlias not apply to values set via pflags?

### DISCUSSION THREAD

**Discussion by shreyb** [spf13/viper#1858] [Q&A]
I have a CLI for which I'm using viper, and I wanted to be able to set a configuration value from either a config file (YAML) or a flag.  Naturally, I figured that I can use an alias so both drive the same value in the `Viper` instance.  However, I'm finding, like in #1598, that the flag being passed seems to have no effect on the underlying value it's aliased to.

The relevant code to reproduce this is something like:

```go
// Set up viper instance
v := viper.New()
v.RegisterAlias("override-flag", "real-flag")

// Set up args
args := []string{"--override-flag"}
f := pflag.NewFlagSet("test", pflag.ContinueOnError)
f.Bool("override-flag", false, "help text")

f.Parse(args)
v.BindPFlags(f)
// At this point, I would expect that since override-flag is passed as a bool pflag, it should change the value of real-flag to true

println(v.GetBool("override-flag"))
println(v.GetBool("real-flag"))	
}
```

The output of both `println` statements is `false`, whereas I'd expect both to be `true`, since the flag `--override-flag` was passed, and is aliased to `real-flag`.  The full example code can be found here:  https://replit.com/@shreyb3/ViperPFlagRegisterAlias.

This is a little bit of a rehash of #1598, because while I agree that that author's solution works to resolve this, this behavior doesn't seem to be explicitly called out in the documentation, so I'm not sure if it's the expected behavior of the library.  Looking at the underlying library code from v1.19.0 (https://github.com/spf13/viper/blob/v1.19.0/viper.go#L1538), it seems that registering an alias does not affect what `viper.Get()` returns for pflag-set values and environment-set configuration values.

**So my question is: Is `RegisterAlias` supposed to not apply to flag-set values?**  

Thanks!

**Comment by sagikazarmark** [comment]
If you move the `RegisterAlias` part after `f.Parse`, it should work. Unfortunately the alias code is quite buggy. :/

**Reply by shreyb** [reply]
Thank you for your reply! Unfortunately, that did not work. Neither did moving the `RegisterAlias` to after `v.BindPFlags`. Do you have any other suggestions? Otherwise, I'm happy to write this up into an issue (and work on a PR to fix it perhaps).
